src/Guide/Api/Methods.hs

```diff
@@ -57,41 +60,49 @@ createCategory db title' group' = do
     Nothing -> do
       catId <- randomShortUid
       time <- liftIO getCurrentTime
-      (_edit, _newCategory) <- dbUpdate db (AddCategory catId title' group' time)
-      -- TODO addEdit edit
+      (_edit, _) <- dbUpdate db (AddCategory catId title' group' time)
+      config <- ask
+      addEdit db config requestDetails _edit
       return catId
 
 -- | Edit categoty's note.
-setCategoryNotes :: DB -> Uid Category -> CTextEdit -> Guider NoContent
-setCategoryNotes db catId CTextEdit{..} = do
+setCategoryNotes :: DB -> RequestDetails -> Uid Category -> CTextEdit -> Guider NoContent
+setCategoryNotes db requestDetails catId CTextEdit{..} = do
   serverModified <- markdownBlockMdSource . _categoryNotes <$> getCategoryOrFail db catId
   checkConflict CTextEdit{..} serverModified
-  _ <- dbUpdate db (SetCategoryNotes catId $ unH cteModified)
+  (_edit, _) <- dbUpdate db (SetCategoryNotes catId $ unH cteModified)
+  config <- ask
+  addEdit db config requestDetails _edit
   pure NoContent
 
 -- | Edit category's info (title, group, status, sections (pro/con, ecosystem, note)).
-setCategoryInfo :: DB -> Uid Category -> CCategoryInfoEdit -> Guider NoContent
-setCategoryInfo db catId CCategoryInfoEdit{..} = do
+setCategoryInfo :: DB -> RequestDetails -> Uid Category -> CCategoryInfoEdit -> Guider NoContent
+setCategoryInfo db requestDetails catId CCategoryInfoEdit{..} = do
   category <- getCategoryOrFail db catId
   -- TODO diff and merge
-  _ <- dbUpdate db $ SetCategoryTitle catId $ unH ccieTitle
-  _ <- dbUpdate db $ SetCategoryGroup catId $ unH ccieGroup
-  _ <- dbUpdate db $ SetCategoryStatus catId $ unH ccieStatus
+  (_editTitle, _) <- dbUpdate db $ SetCategoryTitle catId $ unH ccieTitle
+  (_editGroup, _) <- dbUpdate db $ SetCategoryGroup catId $ unH ccieGroup
+  (_editStatus, _) <- dbUpdate db $ SetCategoryStatus catId $ unH ccieStatus
   let oldEnabledSections = category ^. enabledSections
   let newEnabledSections = unH ccieSections
-  _ <- dbUpdate db $ ChangeCategoryEnabledSections catId
+  (_editSection, _) <- dbUpdate db $ ChangeCategoryEnabledSections catId
       (newEnabledSections S.\\ oldEnabledSections)
       (oldEnabledSections S.\\ newEnabledSections)
-  -- TODO record edits
+  config <- ask
+  mapM_ (addEdit db config requestDetails) [_editTitle, _editGroup, _editStatus, _editSection]
   pure NoContent
 
 -- | Delete a category.
-deleteCategory :: DB -> Uid Category -> Guider NoContent
-deleteCategory db catId = do
+deleteCategory :: DB -> RequestDetails -> Uid Category -> Guider NoContent
+deleteCategory db requestDetails catId = do
   _ <- getCategoryOrFail db catId
-  _mbEdit <- dbUpdate db (DeleteCategory catId)
-  pure NoContent
-  -- TODO mapM_ addEdit mbEdit
+  eitherEdit <- dbUpdate db (DeleteCategory catId)
```

Comment:
I prefer LambdaCase (because of temporal variables) and

```haskell
dbUpdate db (DeleteCategory catId) >>= \case
  Left _ -> pure NoContent
  Right edit' -> do
    config <- ask
    addEdit db config requestDetails edit'
    pure NoContent
```

but for this case there is another way:
```haskell
dbUpdate db (DeleteCategory catId) >>= mapM $ \edit' ->
  config <- ask
  addEdit db config requestDetails edit'
pure NoContent
```